### PR TITLE
Support annotations on enum constants in stub files

### PR DESCRIPTION
**Pull request by jyluo**
Add test suite and fix for https://github.com/typetools/checker-framework/issues/2147

### CODE REVIEW

**Comment by jyluo** [comment]
@kelloggm due to the way CF tests work, I can't seem to just define the enum in 2147 in a separate file in the same test dir (or reuse Issue2156.java) and have it load the annotations from stub. As a simple substitution I use 2 enum constants in java.lang.Thread.State in a stub. Is this ok to you?

I'm waiting for a travis failure before pushing the stub to verify the fix.

**Comment by kelloggm** [comment]
Yes, that seems like a reasonable solution.

File: framework/src/main/java/org/checkerframework/framework/stub/StubParser.java [hunk 1]
```diff
@@ -1281,6 +1309,21 @@ private VariableElement findElement(TypeElement typeElt, VariableDeclarator vari
...
+    /**
+     * Looks for an enum constant element in the typeElt and returns it if the element has the same
+     * name as provided. In case enum constant element is not found it returns null.
+     *
+     * @param typeElt type element where enum constant element should be looked for
+     * @param enumConstDecl the declaration of the enum constant
+     * @return enum constant element in typeElt with the provided name or null if enum constant
+     *     element is not found
+     */
+    private VariableElement findElement(
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/stub/StubParser.java]: How about moving this a bit further up to be next to `private Element findElement(TypeElement typeElt, EnumDeclaration enumDecl)` and keeping the related methods together?

File: framework/src/test/java/tests/StubParserEnumTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package tests;
+
+import java.io.File;
+import java.util.List;
+import org.checkerframework.framework.test.FrameworkPerDirectoryTest;
+import org.junit.runners.Parameterized.Parameters;
+
+public class StubParserEnumTest extends FrameworkPerDirectoryTest {
```
Comment (wmdietl) [inline framework/src/test/java/tests/StubParserEnumTest.java]: I would also move this test to a jtreg test. Many other stub tests are in `checker/jtreg/stubs/`.

**Comment by jyluo** [comment]
Done. I've also adjusted the test in checker/ to use the original example in the issue since in jtreg tests we can pre-compile the enum.

**Pull request merged**